bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -338,6 +338,22 @@ class PlotCanvasView extends Renderer.View
             if is_panning?
               range_info['start'] = rng.get('start')
 
+    min_interval = rng.get('min_interval')
+    max_interval = rng.get('max_interval')
+    if min_interval? || max_interval?
+      new_interval = Math.abs(range_info['end'] - range_info['start'])
+      center = 0.5 * (rng.get('end') + rng.get('start'))
+      sign = if reversed then -1 else 1
+
+      if min_interval > 0
+        if new_interval < min_interval
+          range_info['start'] = center - sign * 0.5 * min_interval
+          range_info['end'] = center + sign * 0.5 * min_interval
+      if max_interval > 0
+        if new_interval > max_interval
+          range_info['start'] = center - sign * 0.5 * max_interval
+          range_info['end'] = center + sign * 0.5 * max_interval
```

Comment:
I would personally refuse to update the range if *either* range constraint would be violated. But if you think otherwise perhaps it's worth a wider discussion.